WordPress/src/main/java/org/wordpress/android/ui/activitylog/list/ActivityLogListFragment.kt

```diff
@@ -100,6 +106,10 @@ class ActivityLogListFragment : Fragment() {
             uiHelpers.updateVisibility(date_range_picker, visibility)
         })
 
+        viewModel.showDateRangePicker.observe(viewLifecycleOwner, Observer { event ->
+            showDateRangePicker(event.initialSelection)
+        })
```

Comment:
I haven't tested it, but I think with this solution the selection won't survive configuration change. Also it'd store the selection even when the user clicked on the "cancel" button, right?

------

I kind of like when the VM layer is the only place which makes any decisions. In this proposal, the view layer knows that we want to show the picker when you click on "date_range_picker" chip - imo it's not responsibility of that layer. The view layer should ideally just propagate user and system events to the VM layer and all the decision should happen there. All this makes testing easier since we can "fake" only system/user events and we can test all the scenarios.

We might also have troubles with tracking - we'd either need to have all tracking in the view layer or we'd need to call something like "viewModel.trackDateRangeChipClicked()" in the on click listener. Which would introduce yet another decision to the view layer.

Wdyt?

> PSS: It all started with me not liking the nullable initialSelection! 😄

Yeah, I don't like that either. We could pass Pair(-1,-1), but that feels even worse.

Reply:
Thank you so much for the detailed response @malinajirka !
> I haven't tested it, but I think with this solution the selection won't survive configuration change. Also it'd store the selection even when the user clicked on the "cancel" button, right?

Actually, you are 💯 right! The suggested by me solution will not survive configuration change, unless more work is done, which I don't think will create a better solution in the end. Also, indeed, the `cancel` button will store the selection, which I don't believe we want, you are right again.

-----

About having the VM layer make all the decisions, I again totally agree. Especially, what you described with tracking clicked. We would anyway need something like that on the VM layer when we will be adding the tracking related code.

> Yeah, I don't like that either. We could pass Pair(-1,-1), but that feels even worse.

I agree, let's keep it as is for now. If I think something better I'll let you know. Passing `Pair(-1, -1) feel worse, I agree.